src/dscanner/analysis/incorrect_infinite_range.d

```diff
@@ -13,85 +13,84 @@ import dparse.lexer;
 /**
  * Checks for incorrect infinite range definitions
  */
-final class IncorrectInfiniteRangeCheck : BaseAnalyzer
+extern(C++) class IncorrectInfiniteRangeCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	mixin AnalyzerInfo!"incorrect_infinite_range_check";
 
 	///
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, null, skipTests);
+		super(fileName);
 	}
 
-	override void visit(const StructBody structBody)
+	override void visit(AST.StructDeclaration sd)
 	{
-		inStruct++;
-		structBody.accept(this);
-		inStruct--;
+		inAggregate++;
+		super.visit(sd);
+		inAggregate--;
 	}
 
-	override void visit(const FunctionDeclaration fd)
+	override void visit(AST.ClassDeclaration cd)
 	{
-		if (inStruct > 0 && fd.name.text == "empty")
-		{
-			line = fd.name.line;
-			column = fd.name.column;
-			fd.accept(this);
-		}
+		inAggregate++;
+		super.visit(cd);
+		inAggregate--;
 	}
 
-	override void visit(const FunctionBody fb)
+	override void visit(AST.FuncDeclaration fd)
 	{
-		if (fb.specifiedFunctionBody && fb.specifiedFunctionBody.blockStatement !is null)
-			visit(fb.specifiedFunctionBody.blockStatement);
-		else if (fb.shortenedFunctionBody && fb.shortenedFunctionBody.expression !is null)
-			visitReturnExpression(fb.shortenedFunctionBody.expression);
-	}
+		import dmd.astenums : Tbool;
 
-	override void visit(const BlockStatement bs)
-	{
-		if (bs.declarationsAndStatements is null)
+		if (!inAggregate)
 			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements is null)
-			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements.length != 1)
-			return;
-		visit(bs.declarationsAndStatements);
-	}
 
-	override void visit(const ReturnStatement rs)
-	{
-		if (inStruct == 0 || line == size_t.max) // not within a struct yet
+		if (!fd.ident || fd.ident.toString() != "empty")
 			return;
-		visitReturnExpression(rs.expression);
-	}
 
-	void visitReturnExpression(const Expression expression)
-	{
-		if (!expression || expression.items.length != 1)
-			return;
-		UnaryExpression unary = cast(UnaryExpression) expression.items[0];
-		if (unary is null)
-			return;
-		if (unary.primaryExpression is null)
+		AST.TypeFunction tf = fd.type.isTypeFunction();
+
+		if (!tf || !tf.next || !tf.next.ty)
 			return;
-		if (unary.primaryExpression.primary != tok!"false")
+
+		AST.ReturnStatement rs = fd.fbody ? fd.fbody.isReturnStatement() : null;
+
+		if (rs)
+		{
+			AST.IntegerExp ie = cast(AST.IntegerExp) rs.exp;
```

Comment:
```suggestion
			AST.IntegerExp ie = rs.exp.isIntegerExp();
```

Reply:
This could be applied if I will be able to get this pr merged in dmd https://github.com/dlang/dmd/pull/14578
`isIntegerExp()` is not implemented in dmd at this moment